samza-core/src/main/java/org/apache/samza/task/TaskCallbackManager.java

```diff
@@ -127,4 +127,14 @@ public List<TaskCallbackImpl> updateCallback(TaskCallbackImpl callback) {
       return ImmutableList.of(callback);
     }
   }
+
+  /**
+   * Override the timeout set in the callback manager with the given new timeout.
+   * This is intended to be used with pipeline drain as we want to override the existing timeout with a higher timeout.
+   *
+   * @param timeout new timeout for process callbacks
+   * */
+  public void updateTaskCallbackTimeout(long timeout) {
```

Comment:
Why are we overriding the existing timeout set as part of the construction of the class? This is confusing and breaks the timeout originally configured that gets passed to the callback manager (which is `task.callback.timeout.ms`).

With this change, we are changing user behavior with setting the default message processing timeout latency to 30 minutes.

Reply:
Changed this.